src/pages/index.js

```diff
@@ -42,24 +124,44 @@ editFormValidator.enableValidation();
 const addFormValidator = new FormValidator(config, addNewCardFormEl);
 addFormValidator.enableValidation();
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (initialCardData) => {
-      const card = createCard(initialCardData);
-      cardSection.prependItem(card);
-    },
-  },
-  ".cards__list"
+const editAvatarFormValidator = new FormValidator(
+  config,
+  profileEditAvatarFormEl
 );
-
-cardSection.renderItems();
+editAvatarFormValidator.enableValidation();
 
 const addNewCardForm = new PopupWithForm("#add-image-popup", (inputValues) => {
-  const cardEl = createCard(inputValues);
-  cardSection.prependItem(cardEl);
+  api.addNewCard(inputValues).then((responseDataObj) => {
+    const newCard = createCard(responseDataObj.data);
+    cardSection.prependItem(newCard);
+  });
 });
 
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-popup",
+  (inputValues) => {
+    editAvatarPopup.setLoadingState(true);
+    api.updateAvatar(inputValues).then(() => {
+      editAvatarPopup.setLoadingState(false);
```

Comment:
Needs correction: Change button state in `finally` block, otherwise the button text won’t change back if an error occurred

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Reply:
corrected